src/ray/core_worker/core_worker.cc

```diff
@@ -76,10 +77,22 @@ ray::JobID GetProcessJobID(const ray::CoreWorkerOptions &options) {
   return options.job_id;
 }
 
+// Helper function converts GetObjectLocationsOwnerReply to ObjectLocation
+std::shared_ptr<ObjectLocation> CreateObjectLocation(
```

Comment:
Nit: Why return a shared pointer? Shouldn't the ownership semantics be determined by the caller?

Reply:
I went though the same question myself! here is my thought process:
- The return value is influenced by the return value of GetLocationFromOwner.
- Since `CoreWorker::GetLocationFromOwner` is used by cython and we want to make the result optional, we have choices between unique_ptr/shared_ptr.  (I don't know  if std::optional works in cython).
- I was preferring unique_ptr over shared_ptr, but I'm particular worried about the case where user call GetLocationFromOwner with duplicate (same) object_ids. using shared_ptr seems make a good trade off in term of memory efficiency (so that results to the same object_id shared the same value).

I think we can make the helper function return by value and deal with shared_ptr in GetLocationFromOwner implementation.
Let me know your thought :)

Reply:
Ah yeah, AFAIK `std::optional`/`absl::optional` isn't supported in Cython (see [issue](https://github.com/cython/cython/issues/3293)), and your reasoning for using a shared pointer over a unique pointer makes sense to me! Thanks for going through your thought process! 😁